assets/js/characterCounter.js

```diff
@@ -0,0 +1,55 @@
+const inputParentElem = document.querySelector(
```

Comment:
regards to this: 
>I know that the govuk-frontend js files provide a character counter already (my code was based on that), but their docs recommend you add all their JS code which I thought was unecessary. Also, their character counter only supports textareas so I thought it would be better if I put my own together.

I had a look at this https://github.com/alphagov/govuk-frontend/blob/e0d5aad300e84e66b42bffa98fa310d630301099/src/govuk/components/character-count/character-count.js

there's nothing to suggest it won't work for text inputs - it's just using the data-module as a selector. If it doesn't work, you can always just change the selector. 

It seems like the project is already including govuk scripts, so really it's no harm in adding another. If you prefer not to, you can just copy + paste this individual script. 

I do think your code is neater (and modern 💪), but unfortunately they do have a more "complete" solution as well as well tested (they are listening for things like `onBlur` as well as handling the plural for word/s or character/s correctly).

Reply:
you could just init the one file - although it seems like they do include some other helpers for accessibility. If you really don't like it you can just copy and paste the one file or snippet

Reply:
Makes sense. I've actually just gone ahead and done that. There are so many polyfill needed for old IE 😢. I remember using the classList one back in the day. Can't believe this browser is still being used.

Reply:
"Still being used" is generous, I believe it's 0.7% global usage at the moment so not much longer

Reply:
really! that small! Why is it still on this list 😭?
https://www.gov.uk/service-manual/technology/designing-for-different-browsers-and-devices#browsers-to-test-in

Reply:
global usage is low - but they do base it off of usage of the gov.uk website, which might have a higher %